modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,110 @@
+package terraform
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/mattn/go-zglob"
+)
+
+type ValidationOptions struct {
+	// The target directory to recursively search for all Terraform directories (those that contain .tf files)
+	// If you provide RootDir and do not pass entries in either IncludeDirs or ExcludeDirs, then all Terraform directories
+	// From the RootDir, recursively, will be validated
+	RootDir string
+	// If you only want to include certain sub directories, add them as paths relative to RootDir. For example, if the
+	// RootDir is /home/project and you want to only include /home/project/examples, add /home/project/examples here
+	// Note that while the struct requires full paths, you can pass relative paths to the NewValidationOptions function
+	// which will build the full paths based on the supplied RootDir
+	IncludeDirs []string
+	// If you want to explicitly exclude certain sub directories, add them as paths relative to RootDir. For example, if the
+	// RootDir is /home/project and you want to include everything EXCEPT /home/project/modules, add
+	// /home/project/modules to this slice. Note that ExcludeDirs is only considered when IncludeDirs is not passed
+	// Note that while the struct requires full paths, you can pass relative paths to the NewValidationOptions function
+	// which will build the full paths based on the supplied RootDir
```

Comment:
```suggestion
	// If you want to explicitly exclude certain sub directories of RootDir, add the absolute paths here. For example, if the
	// RootDir is /home/project and you want to include everything EXCEPT /home/project/modules, add
	// /home/project/modules to this slice. Note that ExcludeDirs is only considered when IncludeDirs is not passed
	// Note that while the struct requires full paths, you can pass relative paths to the NewValidationOptions function
	// which will build the full paths based on the supplied RootDir
```

Reply:
> Note that ExcludeDirs is only considered when IncludeDirs is not passed

Why? The typical idiom with include / exclude is to:

1. Start with a list of included dirs. If the include list passed in by the user is empty, then your initial included dirs list is all subdirs that have `.tf` files.
1. Subtract the excluded dirs.

Reply:
> > Note that ExcludeDirs is only considered when IncludeDirs is not passed
> 
> Why? The typical idiom with include / exclude is to:
> 
>     1. Start with a list of included dirs. If the include list passed in by the user is empty, then your initial included dirs list is all subdirs that have `.tf` files.
> 
>     2. Subtract the excluded dirs.

This is actually how it works, so I think my description is confusing: 

1. Default behavior (if you pass neither IncludeDirs nor ExcludeDirs) is that all subdirs that have `.tf` files are found 
2. If you pass IncludeDirs, all subdirs that have `.tf` files are found, and only the intersection of IncludeDirs and all subdirs that have `.tf` files is returned. 
3. If you pass ExcludeDirs, all the subdirs that have `.tf` files are found, minus those included in ExcludeDirs. 

My main point is that if you for some reason pass IncludeDirs and ExcludeDirs, the code will return after processing your IncludeDirs. In other words, you should pass one or the other to achieve your desired targeting behavior.

Reply:
I'll update the comments to be more clear on this point.

Reply:
I think what you have is:

```
foldersToProcess = terraformModulesIn(rootDir)

if includeDirs
  return intersection(foldersToProcess, includeDirs)
else if excludeDirs
  return difference(foldersToProcess, excludeDirs)
else 
  return foldersToProcess
```

The more idiomatic include / exclude implementation is a bit different:

```
foldersToProcess = terraformModulesIn(rootDir)

if includeDirs
  foldersToProcess = intersection(foldersToProcess, includeDirs)

if excludeDirs
  foldersToProcess = difference(foldersToProcess, excludeDirs)

return foldersToProcess
```

Reply:
Gotcha. Thanks for the explanation. Fixed in af208cc